utils/retry/retryexecutor.go

```diff
@@ -0,0 +1,58 @@
+package retry
+
+import (
+	"fmt"
+	"github.com/jfrog/jfrog-client-go/utils/log"
+	"time"
+)
+
+const retryMessage = "%sAttempt %v - %s"
+
+type ExecutionHandlerFunc func() (bool, error)
+
+type RetryExecutor struct {
+	// The amount of retries to perform.
+	MaxRetries int
+
+	// Time period between retries.
```

Comment:
// Time period between retries. --> // Number of seconds to sleep between retries.